web/src/routes/(user)/utilities/duplicates/[[photos=photos]]/[[assetId=id]]/+page.svelte

```diff
@@ -11,42 +11,124 @@
-      await deleteAssets({ assetBulkDeleteDto: { ids: trashIds, force: !$featureFlags.trash } });
-
-      data.duplicates = data.duplicates.filter((duplicate) => duplicate.duplicateId !== duplicateId);
+  const withConfirmation = async (callback: () => Promise<void>, prompt?: string, confirmText?: string) => {
+    if (prompt && confirmText) {
+      const isConfirmed = await dialogController.show({
+        prompt: $t(prompt),
+        confirmText: $t(confirmText),
+      });
 
-      if (trashIds.length === 0) {
+      if (!isConfirmed) {
         return;
       }
+    }
 
-      notificationController.show({
-        message: $t('assets_moved_to_trash', { values: { count: trashIds.length } }),
-        type: NotificationType.Info,
-      });
+    try {
+      return await callback();
     } catch (error) {
       handleError(error, $t('errors.unable_to_resolve_duplicate'));
     }
   };
+
+  const deletedNotification = (trashedCount: number) => {
+    if (!trashedCount) {
+      return;
+    }
+
+    notificationController.show({
+      message: $t($featureFlags.trash ? 'assets_moved_to_trash' : 'assets_deleted', {
+        values: { count: trashedCount },
+      }),
+      type: NotificationType.Info,
+    });
+  };
+
+  const handleResolve = async (duplicateId: string, duplicateAssetIds: string[], trashIds: string[]) => {
+    return withConfirmation(
+      async () => {
+        await deleteAssets({ assetBulkDeleteDto: { ids: trashIds, force: !$featureFlags.trash } });
+        await updateAssets({ assetBulkUpdateDto: { ids: duplicateAssetIds, duplicateId: null } });
+
+        data.duplicates = data.duplicates.filter((duplicate) => duplicate.duplicateId !== duplicateId);
+
+        deletedNotification(trashIds.length);
+      },
+      trashIds.length > 0 && !$featureFlags.trash ? 'delete_duplicates_confirmation' : undefined,
+      trashIds.length > 0 && !$featureFlags.trash ? 'permanently_delete' : undefined,
+    );
+  };
+
+  const handleDeduplicateAll = async () => {
+    return withConfirmation(
+      async () => {
+        const idsToKeep = data.duplicates
+          .map((group) => suggestDuplicateByFileSize(group.assets))
+          .map((asset) => asset?.id);
+        const idsToDelete = data.duplicates.flatMap((group, i) =>
+          group.assets.map((asset) => asset.id).filter((asset) => asset !== idsToKeep[i]),
+        );
+
+        await deleteAssets({ assetBulkDeleteDto: { ids: idsToDelete, force: !$featureFlags.trash } });
+        await updateAssets({
+          assetBulkUpdateDto: {
+            ids: [...idsToDelete, ...idsToKeep.filter((id): id is string => !!id)],
+            duplicateId: null,
+          },
+        });
+
+        data.duplicates = [];
+
+        deletedNotification(idsToDelete.length);
+      },
+      $featureFlags.trash ? 'bulk_trash_duplicates_confirmation' : 'bulk_delete_duplicates_confirmation',
```

Comment:
Maybe we can provide the exact count to the user

Reply:
Good point, I can see that being helpful.

Reply:
Yeah, definitely much better now.